x-pack/plugins/data_visualizer/public/application/common/components/document_count_content/document_count_content.tsx

```diff
@@ -30,14 +77,115 @@ export const DocumentCountContent: FC<Props> = ({ documentCountStats, totalCount
     chartPoints = Object.entries(buckets).map(([time, value]) => ({ time: +time, value }));
   }
 
+  const approximate = documentCountStats.randomlySampled === true;
+
+  const ProbabilityUsed =
+    randomSamplerPreference !== RANDOM_SAMPLER_OPTION.OFF && isDefined(samplingProbability) ? (
+      <FormattedMessage
+        id="xpack.dataVisualizer.randomSamplerSettingsPopUp.probabilityLabel"
+        defaultMessage="Probability used: {samplingProbability}%"
+        values={{ samplingProbability: samplingProbability * 100 }}
+      />
+    ) : null;
   return (
     <>
-      <TotalCountHeader totalCount={totalCount} />
+      <EuiFlexGroup alignItems="center" gutterSize="xs">
+        <TotalCountHeader totalCount={totalCount} approximate={approximate} loading={loading} />
+        <EuiFlexItem grow={false}>
+          <EuiPopover
+            id="dscSamplingOptions"
+            button={
+              <EuiToolTip
+                content={i18n.translate('xpack.dataVisualizer.samplingOptionsButton', {
+                  defaultMessage: 'Sampling options',
+                })}
+              >
+                <EuiButtonIcon
+                  size="xs"
+                  iconType="gear"
+                  onClick={onShowSamplingOptions}
+                  data-test-subj="discoverSamplingOptionsToggle"
+                  aria-label={i18n.translate('xpack.dataVisualizer.samplingOptionsButton', {
+                    defaultMessage: 'Sampling options',
+                  })}
+                />
+              </EuiToolTip>
+            }
+            isOpen={showSamplingOptionsPopover}
+            closePopover={closeSamplingOptions}
+            panelPaddingSize="none"
+            anchorPosition="downLeft"
+          >
+            <EuiPanel style={{ maxWidth: 400 }}>
+              <EuiFlexItem grow={true}>
+                <EuiCallOut
+                  size="s"
+                  color={'primary'}
+                  title={i18n.translate(
+                    'xpack.dataVisualizer.randomSamplerSettingsPopUp.infoCalloutMessage',
+                    {
+                      defaultMessage:
+                        'Random sampler is being used for the total document count and the chart. Pick a higher percentage for better accuracy, or "Off" for no sampling.',
```

Comment:
I think this message should change depending on what option you have selected. For example, if it is set to `Off`, say that random sampling can be turned on for the total document count and chart to increase performance although some accuracy will be lost.

If set to `On - automatic`, then something like, Random sampling is being used for the total document count and the chart. The probability used in the aggregation will be automatically set to balance accuracy and speed.

If set to `On - manual`, then something like, Random sampling is being used for the total document count and the chart. A lower percentage probability will increase performance, but some accuracy will be lost.

Reply:
I agree, if it's possible to customize those messages, it would be great!